App.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import Settings from './client/src/Components/SettingsPage/Settings'
+import { NavigationContainer } from '@react-navigation/native'
+import { createNativeStackNavigator } from '@react-navigation/native-stack'
+import WelcomingScreen from './client/src/Components/WelcomingPage/WelcomingScreen'
+import Login from './client/src/Components/LoginPage/Login'
+import CreateAccount from './client/src/Components/SignUpPage/CreateAccount'
+import ForgotPassword from './client/src/Components/LoginPage/ForgotPassword'
+import UserProfile from './client/src/Components/Profile/UserProfile'
+import HomeNewsScreen from './client/src/Components/HomePage/HomeNewsScreen'
+import AddNewsPopUp from './client/src/Components/HomePage/NewsPopUp/AddNewsPopUp'
+import { observer } from 'mobx-react-lite'
+import getAuthStore from './client/src/stores/authenticationStore'
+
+
+
+const App : React.FC =  observer(()=> {
+  const Stack = createNativeStackNavigator();  
+
+
+  return (
+    <NavigationContainer>
+      <Stack.Navigator initialRouteName= { !(getAuthStore().isAuthenticated.get()) ? "HomeNewsScreen" : "Welcoming" }>
```

Comment:
usually it isn't a good idea to rely on the store which is still not instantiated in order to set the initial route name, u can use mmkv for this case

Reply:
Mr Anthony, I have understood your point but if you see the authentication store , when setting the isAuthenticated to true , I am using mmkv storage , I am getting the "authToken" after setting it in the Login Function, I mean If the user Logged in , it will set the mmkv ,  and will set the isAuthenticated observable to true so we are not relying on the store directley , we are relying on the mmkv storage